scipy/spatial/transform/rotation.py

```diff
@@ -1,12 +1,55 @@
 from __future__ import division, print_function, absolute_import
 
+__all__ = ['Rotation']
+
 import numpy as np
+import scipy.linalg
 
 class Rotation(object):
     """Rotation in 3 dimensions.
 
     This class will include initializers from different representations, converters
     and some useful algorithms such as quaternion slerp and rotation estimation.
+
     """
-    def __init__(self):
-        pass
+    def __init__(self, quaternions):
+        """Initialize class from possible normalized quaternions.
+
+        Parameters
+        ----------
+        quaternions : (N, 4) numpy.array
+                      Each row is a quaternion stored in scalar-last
+                      (x, y, z, w) format.
+
+        """
+        # Each row should have 4 numbers
```

Comment:
So, first you need to convert using `np.asarray`, then check it's shape and raise `ValueError` if something is wrong with it.